<launch>
    <!-- <param name="use_sim_time" value="true" /> -->
    <node name="rplidarNode" pkg="rplidar_ros" type="rplidarNode" output="screen">
    <param name="serial_port"         type="string" value="/dev/ttyUSB0"/>  
    <param name="serial_baudrate"     type="int"    value="115200"/>
    <param name="frame_id"            type="string" value="laser"/>
    <param name="inverted"            type="bool"   value="false"/>
    <param name="angle_compensate"    type="bool"   value="true"/>
    </node>
    
    <node name="rosserial_node" pkg="rosserial_python" type="serial_node.py" output="screen">
    <param name="port"         type="string" value="/dev/ttyACM0"/>
    <param name="baud"         type="int"    value="57600"/>
    </node>

    <node pkg="tf" type="static_transform_publisher" name="transform" 
      args= "0 0 0 0 0 0 map laser 100" output="screen"></node> 

    <node pkg="hector_mapping" type="hector_mapping" name="slam_hector" output="screen">
      <param name="pub_map_odom_transform" value="true"/>
      <param name="base_frame" value="laser"/>
      <param name="odom_frame" value="laser"/>
      <param name="map_frame"  value="map"/>
      <param name="map_update_distance_thresh" value="0.1"/>
      <param name="map_update_angle_thresh" value="0.1" />
    </node>
    <node name="follow_wall_node" pkg="follow_wall" type="follow_wall_node" output="screen" />
    <node name="stateaction_map_node" pkg="follow_wall" type="stateaction_map_node.py" output="screen" />

</launch>